### ISSUE TITLE
Can't perform a React state update on an unmounted component when performing a http request with useForm methods

### ISSUE THREAD

**Issue by amiranagram** [inertiajs/inertia#656]
### Versions:

- `@inertiajs/inertia` version: 0.8.7
- `@inertiajs/inertia-react` version: 0.5.12

### Describe the problem:

```
Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

I'm a noobie with react (coming from vue) so bear with me. After some digging I found out that this error happens when trying to modify the state inside a component after the component has been unmounted. In my case after submitting a post request with `post` method from Inertia's `useForm`.

First I thought I messed up something (react noob) then I cloned [pingcrm-react](https://github.com/Landish/pingcrm-react) and the same thing happens there.

### Steps to reproduce:

Have a form that and use `useForm` to manage the state and submit it.

Ex:

```
const {data, setData, post, processing} = useForm({
    email: '',
    password: '',
    remember: false,
});

function submit(e) {
    e.preventDefault();

    post(route('login'));
}
```

Or you can clone [pingcrm-react](https://github.com/Landish/pingcrm-react), set up the project and try to login. After successful login when it navigates away from the login page you'll see the error in console.

Hope I was clear enough.

**Comment by mrtijn** [comment]
Got the same issue. Using `Inertia.post` instead of the `post` from `useForm` seems to fix it.

**Comment by reinink** [comment]
This has been fixed in #680 and will be released soon (hopefully tomorrow). 👍
